iac-common/src/main/java/org/sonar/iac/common/extension/visitors/SyntaxHighlightingVisitor.java

```diff
@@ -57,7 +58,10 @@ protected void after(InputFileContext ctx, Tree root) {
     newHighlighting.save();
   }
 
-  protected void highlight(HasTextRange tree, TypeOfText typeOfText) {
+  protected void highlight(@Nullable HasTextRange tree, TypeOfText typeOfText) {
+    if (tree == null) {
```

Comment:
The condition is covered by `ArmHighlightingVisitor:62-63`

Reply:
Why the tree can be null? And when this only comes from the Arm highlighter, it might be smarter to catch null in the dedicated visitor it self.